src/main/groovy/io/openliberty/tools/gradle/tasks/GenerateFeaturesTask.groovy

```diff
@@ -399,86 +257,73 @@ class GenerateFeaturesTask extends AbstractFeatureTask {
+    private getMPVersion(Object project) {
+        String mpVersion = null
+        project.configurations.compile.allDependencies.each {
+            if (it.group.equals("org.eclipse.microprofile") &&
+                it.name.equals("microprofile")) {
+                if (it.version.startsWith("1")) {
+                    mpVersion =  "mp1"
+                } else if (it.version.startsWith("2")) {
+                    mpVersion =  "mp2"
+                } else if (it.version.startsWith("3")) {
+                    mpVersion =  "mp3"
+                } else if (it.version.startsWith("4")) {
+                    mpVersion =  "mp4"
                 }
             }
         }
-        return features;
+        return mpVersion;
     }
 
-    // A class to pass the list of conflicts back to the caller.
-    private class NoRecommendationException extends Exception {
-        private static final long serialVersionUID = 1L;
-        Set<String> conflicts;
-        NoRecommendationException(Set<String> conflictSet) {
-            conflicts = conflictSet;
+    // Define the logging functions of the binary scanner handler and make it available in this plugin
+    private class BinaryScannerHandler extends BinaryScannerUtil {
+        BinaryScannerHandler(File scannerFile) {
+            super(scannerFile);
         }
-        public Set<String> getConflicts() {
-            return conflicts;
+        @Override
+        public void debug(String msg) {
+            logger.debug(msg);
         }
-    }
-
-    // A class that encapsulates a list of conflicting features, a suggested list of replacements
-    // and a flag that indicates whether the conflicts were found in the features existing in the
-    // app's server config or if the conflicts exist in the binary files we examined.
-    private class RecommendationSetException extends Exception {
-        private static final long serialVersionUID = 1L;
-        boolean existingFeaturesConflict;
-        Set<String> conflicts;
-        Set<String> suggestions;
-        RecommendationSetException(boolean existing, Set<String> conflictSet, Set<String> suggestionSet) {
-            existingFeaturesConflict = existing;
-            conflicts = conflictSet;
-            suggestions = suggestionSet;
+        @Override
+        public void debug(String msg, Throwable t) {
+            logger.debug(msg, t);
         }
-        public boolean isExistingFeaturesConflict() {
-            return existingFeaturesConflict;
+        @Override
+        public void error(String msg) {
+            logger.error(msg);
         }
-        public Set<String> getConflicts() {
-            return conflicts;
+        @Override
+        public void warn(String msg) {
+            logger.warn(msg);
         }
-        public Set<String> getSuggestions() {
-            return suggestions;
+        @Override
+        public void info(String msg) {
+            logger.info(msg);
```

Comment:
We may want to change this to `logger.lifecycle(msg);` similar to DevTask https://github.com/OpenLiberty/ci.gradle/blob/1bca3270867ddfffa0e8654479dd420b0764c363/src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy#L327 so that Gradle will show these by default.

Reply:
👍